server/pkg/schema/schema.go

```diff
@@ -30,5 +26,42 @@ func (s *Schema) AddField(f Field) {
 	if s.fields == nil {
 		s.fields = []*Field{}
 	}
+	for _, sf := range s.fields {
+		if sf.id.Equal(f.ID()) {
+			return
+		}
+	}
```

Comment:
We have generics!

```go
if lo.SomeBy(s.fields, func(g *Field) bool { return g.ID().Equal(f.ID()) }) {
  return
}
```

Furthermore, you are good at refactoring if you can come up with this much.

```go
func (s *Schema) HasField(f FieldID) bool {
  return lo.SomeBy(s.fields, func(g *Field) bool { return g.ID().Equal(f) })
}

// ...

if s.HasField(f.ID()) {
  return
}
```

Instead of mixing various processes with different meanings in a single function, it is better to name them and separate them for better visibility.

Reply:
Thanks for this comment, I need to be more careful